charts/rancher-istio/101.1.0+up1.15.3/configs/istio-base.yaml

```
@@ -19,6 +19,10 @@ spec:
     - enabled: {{ .Values.egressGateways.enabled }}
       name: istio-egressgateway
       k8s:
+        hpaSpec:
+          minReplicas: {{ .Values.egressGateways.minReplicas }}
+        podDisruptionBudget:
+          minAvailable: {{ .Values.egressGateways.podDisruptionBudget.minAvailable }}
```

Is there any reason why we want to expose only these specific fields? IMO, we should just allow users to provide the blanket `.Values.*.hpaSpec` and `.Values.*.podDistruptionBudget` they desire:

```suggestion
{{- if .Values.egressGateways.hpaSpec }}
        hpaSpec: {{ toYaml .Values.egressGateways.hpaSpec | nindent 10 }}
{{- end }}
{{- if .Values.egressGateways.podDisruptionBudget }}
        podDisruptionBudget: {{ toYaml .Values.egressGateways.podDisruptionBudget | nindent 10 }}
{{- end }}
```

---

@Tejeev thoughts here?

---

this suggestion seems good to me
updated the PR with suggested changes.

---

Discussed with Fred as well and none of us can think of any reason not to open up these HA management configurations to the users. I know I've seen at least one customer (deploying Istio independently) that managed their Istio replica counts